instrumentation/grpc-1.6/library/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/TracingServerInterceptor.java

```diff
@@ -127,29 +128,28 @@ public void onCancel() {
           Span.fromContext(context).setAttribute("grpc.canceled", true);
         }
       } catch (Throwable e) {
-        tracer.endExceptionally(context, e);
+        instrumenter.end(context, request, null, e);
         throw e;
       }
-      tracer.end(context);
+      instrumenter.end(context, request, null, null);
     }
 
     @Override
     public void onComplete() {
       try {
         delegate().onComplete();
       } catch (Throwable e) {
-        tracer.endExceptionally(context, e);
+        instrumenter.end(context, request, null, e);
         throw e;
       }
-      tracer.end(context);
```

Comment:
I think ending in `close` handles this, either that or we don't have any unit test covering this case